keras/utils/data_utils.py

```diff
@@ -282,3 +295,322 @@ def validate_file(fpath, file_hash, algorithm='auto', chunk_size=65535):
         return True
     else:
         return False
+
+
+class Sequence(object):
+    """ Base object for fitting to a sequence of data, such as a dataset.
+    Every `Sequence` must implements the `__getitem__` and the `__len__` methods.
+
+    # Examples
+    ```python
+    from skimage.io import imread
+    from skimage.transform import resize
+    import numpy as np
+
+    # Here, `x_set` is list of path to the images
+    # and `y_set` are the associated classes.
+
+    class CIFAR10Sequence(Sequence):
+    class CIFAR10Sequence(Sequence):
+        def __init__(self, x_set, y_set, batch_size):
+            self.X,self.y = x_set,y_set
+            self.batch_size = batch_size
+
+        def __len__(self):
+            return len(self.X) // self.batch_size
+
+        def __getitem__(self,idx):
+            batch_x = self.X[idx*self.batch_size:(idx+1)*self.batch_size]
+            batch_y = self.y[idx*self.batch_size:(idx+1)*self.batch_size]
+
+            return np.array([resize(imread(file_name), (200,200))
+                                for file_name in batch_x]),
+             np.array(batch_y)
```

Comment:
Pretty hard to read, put the resize on a separate line, not on the `return` line